packages/hoppscotch-selfhost-web/src/platform/collections/collections.platform.ts

```diff
@@ -828,3 +842,105 @@ function getRequestIndex(
 
   return requestIndex
 }
+
+function issueBackendIDToDuplicatedCollection(
+  collectionStore: ReturnType<
+    typeof getStoreByCollectionType
+  >["collectionStore"],
+  collectionType: ReqType,
+  userCollectionBackendID: string,
+  parentCollectionID?: string
+): boolean {
```

Comment:
This function is intended to assign the backend-issued ID received from the GQL subscription to a new collection in the existing store. When duplicating a collection, the new entry is given an ID that follows the value from the source collection, with a `-duplicate-collection` suffix.

For actions other than duplication, a new entry will be added to the store, as there won't be a collection with such an ID existing in the store. Adjustments will be made to the syncing logic based on the new dedicated GQL subscription for collection duplication actions, which will send the entire tree.